mtp_noms_ops/apps/security/tests/test_views.py

```diff
@@ -5340,6 +5343,114 @@ def test_resolve_page_displays_error_if_assignment_collision_from_accept_reject(
+                    ]
+                }
+                for i in range(page_size)
+            ]
+        }
+
+    @staticmethod
+    def _only_active_states(states):
+        return list(filter(lambda x: x['active'], states))
+
+    def test_cannot_access_view(self):
+        """
+        Test that if the logged in user doesn't have the right permissions, he/she
+        gets redirected to the dashboard.
+        """
+        user_data = self.get_user_data(permissions=required_permissions)
+        with responses.RequestsMock() as rsps:
+            self.login(rsps=rsps, user_data=user_data)
+            response = self.client.get(reverse('security:auto_accept_rule_list'), follow=True)
+            self.assertRedirects(response, reverse('security:dashboard'))
+
+    def test_view(self):
+        """
+        Test that the view displays all auto-accepts returned by the API.
+        """
+        api_auto_accept_response_len = 50
+        page_size = SECURITY_FORMS_DEFAULT_PAGE_SIZE
+        api_auto_accept_response = self._generate_auto_accept_response(api_auto_accept_response_len, page_size)
+
+        with responses.RequestsMock() as rsps:
+            self.login(rsps)
+            self.mock_need_attention_count(rsps, 0)
+            rsps.add(
+                rsps.GET,
+                api_url('/security/checks/auto-accept/'),
+                json=api_auto_accept_response
+            )
+
+            response = self.client.get(reverse('security:auto_accept_rule_list'), follow=True)
+
+            content = response.content.decode()
+            for resp in api_auto_accept_response['results']:
+                self.assertIn(f'************{resp["debit_card_sender_details"]["card_number_last_digits"]}', content)
+                self.assertIn(resp['debit_card_sender_details']['cardholder_names'][0], content)
+                self.assertIn(resp['prisoner_profile']['prisoner_number'], content)
+                self.assertIn(resp['prisoner_profile']['prisoner_name'], content)
+                self.assertIn(
+                    '{last_name}, {first_name}'.format(
+                        last_name=self._only_active_states(resp['states'])[-1]['added_by']['last_name'],
+                        first_name=self._only_active_states(resp['states'])[-1]['added_by']['first_name']
+                    ),
+                    content
+                )
+                self.assertIn(
+                    parser.isoparse(
+                        self._only_active_states(resp['states'])[-1]['created']
+                    ).strftime('%d/%m/%Y %H:%M'),
+                    content
+                )
+                self.assertIn(
+                    f'Showing 1 to {page_size} of {api_auto_accept_response_len} auto accept rules',
+                    content
+                )
+                self.assertIn(
+                    f'{api_auto_accept_response_len} auto accept rules',
+                    content
+                )
+
```

Comment:
might be worth having a test that checks the ordering params are passed up to the api. just something simple given "i am sorting by column X" therefore "param_x=y" is passed to the api

Reply:
If I understand you correctly, the test would be checking that the ordering parameter was passed correctly to the HTTP request to the API service, by making a similar call to noms-ops as would be made by `sorted_cell`?

Reply:
yes that's exactly what i would suggest. right now, i don't think it does because the form only has a page field right now